src/map/Node.kt

```diff
@@ -1,6 +1,10 @@
+package map
+
+import agent.Animal
+import agent.Teacher
 import java.util.HashSet
 
-class NodeContent {
+class Node(val xCoordinate: Int, val yCoordinate: Int) {
     val animals = HashSet<Animal>()
     val teachers = HashSet<Teacher>()
     val isBlockingNode: Boolean = false
```

Comment:
Just an idea for the future is to implement a class hierarchy of different types of nodes (wall node, etc), and offer a visitor interface for easier processing.